### PR TITLE
Fix/0025604/6 0/remove last visited entries

### PR DESCRIPTION
**Pull request by chfsx**
@alex40724 I need some changes in `ilNavigationHistoryGUI` for Mantis https://mantis.ilias.de/view.php?id=25604
The new GS-Items in the MainMenu do not accept any "async" links, therefore I change the behaviour to a sync-request.
If OK, I cherry-pick this to 5.4 as well

### CODE REVIEW

File: Services/Navigation/classes/class.ilNavigationHistoryGUI.php [hunk 1]
```diff
@@ -152,13 +158,15 @@ function handleNavigationRequest()
...
-	function removeEntries()
+	public function removeEntries()
 	{
 		$ilNavigationHistory = $this->nav_history;
-		
+
 		$ilNavigationHistory->deleteDBEntries();
 		$ilNavigationHistory->deleteSessionEntries();
+
+		$script = $this->http->request()->getHeader('Referer')[0];
```
Comment (chfsx) [inline Services/Navigation/classes/class.ilNavigationHistoryGUI.php:168]: @alex40724 the magic is here, after removing the entries I just redirect to the last location.

Comment (alex40724) [reply Services/Navigation/classes/class.ilNavigationHistoryGUI.php:168]: Won't this fail on POST requests (commands not part of the Referer)?

**Comment by alex40724** [comment]
@chfsx Thanks for the PR. I have only on question above.

And sorry for the late feedback, I just saw this is a few months old. I rarely filter for "waiting for my review" here...

File: Services/Navigation/classes/class.ilNavigationHistoryGUI.php (hunk 1 above)
Comment (chfsx) [reply Services/Navigation/classes/class.ilNavigationHistoryGUI.php:168]: valid point... there are a lot of those situations in ILIAS. did not consider that. I‘ll look for another solution then

**Pull request closed without merging**